External/FEXCore/include/FEXCore/Utils/DeferredSignalMutex.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
```

Comment:
Did you move this file? Looks like the previous one wasn't removed, so this shows up as new code.

Reply:
This file contains objects that behave differently than the other `ScopedSignalMask` and relies on `FEXCore::Core::InternalThreadState` so it can't be in the FHU folder, but is mostly a drop-in replacement barring some nasty edges.

Reply:
Creating an ever-increasing number of copies of the already poorly-made ScopedSignalMask abstraction is becoming a maintenance burden, considering how widely it's used and how difficult it has been for us to actually pin down its purpose. The variants added in this PR add yet more complexity and throw 3 different things together, and it's a sign we're heading the wrong way that we need more than 40 characters to come up with even vaguely descriptive names anymore.

After some thinking, I've drafted a counter-proposal in #2691 that changes the interfaces so that we can use `std::unique_lock`/`std::shared_lock` instead. To cover the needs of this PR, that design just needs a scope guard added to it to handle refcounting.

Does this new design suit you? We can merge this PR as-is and adopt the new design later if yes, otherwise we should discuss how to handle the added ScopedSignalMask requirements in a more workable manner.

Reply:
Sure, that design works. Would love to get this in so this three month old PR can finally be done with.